src/upgrades/mod.rs

```diff
@@ -480,87 +773,103 @@ pub fn finalise_data_migration(
+                PUBSERVER_CONTENT_NS,
+                PUBSERVER_NS,
+                SIGNERS_NS,
+                STATUS_NS,
+                TA_PROXY_SERVER_NS,
+                TA_SIGNER_SERVER_NS,
+            ] {
+                // Data structure is as follows:
+                //
+                //   data_dir/
+                //            upgrade-data/      --> upgraded (may be missing)
+                //                         ns1,
+                //                         ns2,
+                //                         etc
+                //             ns1, --> current
+                //             ns2,
+                //             etc
+                //
+                //             archive-prev-v/   --> archived current dirs which were upgraded
+                //
+                let upgraded_dir = upgrade_base_dir.join(ns.as_str());
+                let archive_dir = archive_base_dir.join(ns.as_str());
+                let current_dir = data_dir.join(ns.as_str());
+
+                if upgraded_dir.exists() {
+                    // Data was prepared. So we archive the current data and
+                    // then move the prepped data.
+                    move_dir(&current_dir, &archive_dir)?;
+                    move_dir(&upgraded_dir, &current_dir)?;
+                } else if current_dir.exists() {
+                    // There was no new data for this directory. But, we make
+                    // make a backup so that we can have a consistent data set
+                    // to fall back to in case of a downgrade.
+                    file::backup_dir(&current_dir, &archive_dir).map_err(|e| {
+                        Error::Custom(format!(
+                            "Could not backup directory {} to {} after migration: {}",
+                            current_dir.to_string_lossy(),
+                            archive_dir.to_string_lossy(),
+                            e
+                        ))
+                    })?;
+                }
 
-        let pubd = data_dir.join(PUBSERVER_NS.as_str());
-        let pubd_arch = data_dir.join(format!("arch-{}-{}", PUBSERVER_NS.as_str(), from));
-        let pubd_upg = upgrade_dir.join(PUBSERVER_NS.as_str());
-        let pubd_objects = data_dir.join(PUBSERVER_CONTENT_NS.as_str());
-        let pubd_objects_arch = data_dir.join(format!("arch-{}-{}", PUBSERVER_CONTENT_NS.as_str(), from));
-        let pubd_objects_upg = upgrade_dir.join(PUBSERVER_CONTENT_NS.as_str());
-
-        // upgrade-data/pubd exists
-        if pubd_upg.exists() {
-            // pubd -> arch-pubd-{old-version}
-            // upgrade-data/pubd -> pubd
-            move_dir_if_exists(&pubd, &pubd_arch)?;
-            move_dir_if_exists(&pubd_upg, &pubd)?
-        }
+                let version_file = current_dir.join("version");
+                if version_file.exists() {
+                    debug!("Removing excess version file: {}", version_file.to_string_lossy());
```

Comment:
Is it old or excess, you refer to it differently in these two log messages.

Reply:
fixed